scripts/src/main/resources/scripts/command/python

```diff
@@ -0,0 +1,186 @@
+      PYTHON_VERSION=$(curl -L https://www.python.org/ftp/python 2>&1 | grep -o '3\.[0-9]*\.[0-9]*' | sort -u -V | tail -n2 | head -n1)
+	  fi
+
+    if doIsMacOs
+    then
+      doInstall "-" "${PYTHON_HOME}" "python" "${PYTHON_VERSION}" "" "Python"
+      sh "${PYTHON_HOME}"/configure --enable-optimizations --with-ensurepip=install --prefix="${PYTHON_HOME}"
+      make -j 8
+      make altinstall
+      
+    elif doIsWindows
+    then
+      doEcho "Getting pip install file"
+      # https://github.com/pypa/get-pip
+      curl -sSL https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+      doInstall "-" "${PYTHON_HOME}" "python" "${PYTHON_VERSION}" "" "python"
+      doRunPython "get-pip.py"
+      rm get-pip.py
+      if [ -a "$(find "${PYTHON_HOME}" -name '*._pth')" ]
+      then
+        PTH_FILE="$(find "${PYTHON_HOME}" -name '*._pth')"
+        doEcho "Freeing Python from embedded Mode..."
+        sed -i -e 's/#import site/import site/g' "${PTH_FILE}"
+      else
+        doEcho "Error. Couldn't find _pth file."
+      fi
+    else
+      # install build prerequisites      
+      pckmngers=("apt-get" "yum" "zypp")
+
+      for t in "${pckmngers[@]}"
+      do
+        command -v "${t}" > /dev/null
+        if [ "${?}" -eq "0" ]
+        then
+          pckmnger=${t}
+        fi
+      done
+      
+      case ${pckmnger} in
+        apt-get) doEcho "installing required dependencies..."
+                 doRunCommand "sudo ${pckmnger} install -y make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev";;
+        yum)     doEcho "installing required dependencies..."
+                 doRunCommand "sudo ${pckmnger} -y groupinstall \"Development Tools\""
+                 doRunCommand "sudo ${pckmnger} -y install gcc openssl-devel bzip2-devel libffi-devel";;
+        zypp)    doEcho "installing required dependencies..."
+                 doRunCommand "sudo ${pckmnger} install automake fdupes gcc gcc-c++ gcc-fortran gdbm-devel gettext-tools gmp-devel intltool libbz2-devel libexpat-devel libffi-devel libnsl-devel lzma-devel make ncurses-devel netcfg openssl-devel pkgconfig readline-devel sqlite-devel xz zlib-devel";;
+        *)       doEcho "Sorry, your package manager is not supported."
+                 exit;;
+      esac
+
+      doInstall "-" "${PYTHON_HOME}" "python" "${PYTHON_VERSION}" "" "Python"
+      "${PYTHON_HOME}"/configure --enable-optimizations --with-ensurepip=install --prefix="${PYTHON_HOME}"
+      make -j 8
+      make altinstall
+    fi
+    # rename folder from bin to Scripts
+    if [ -d "${PYTHON_HOME}/bin" ]
+    then
+      mv "${PYTHON_HOME}"/bin "${PYTHON_HOME}"/Scripts
+    fi
+  fi
+
+  if [ "${1}" != "silent" ] && ! doIsQuiet
+  then
+    doRunPython "--version"
+  fi 
+  
+  setPath
+
+}
+
+function setPath() {
+  py_dir="${PYTHON_SCRIPTS}"
+  path_bind="PATH=\$PATH:"
+  set_path="${path_bind}${py_dir}"
```

Comment:
Unfortunately by default variables are global in bash (historically grown).
We should always use `local` keywords and only use global variables in functions where this is explicitly desired (what is extremely rare).

Reply:
Thanks for improving. Done.